client-v2/src/components/FrontsEdit/FrontContainer.tsx

```diff
@@ -152,31 +159,33 @@ class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
           <FrontHeader greyHeader={true}>
             <FrontsHeaderText title={title}>{title}</FrontsHeaderText>
             <FrontHeaderMeta>
-              <a
-                href={`https://preview.gutools.co.uk/responsive-viewer/https://preview.gutools.co.uk/${
-                  this.props.frontId
-                }`}
-                target="preview"
-              >
-                <FrontHeaderButton size="l">
-                  <PreviewEyeIcon size="xl" />
-                  <PreviewButtonText>Preview</PreviewButtonText>
-                </FrontHeaderButton>
-              </a>
-              <StageSelectButtons>
-                <RadioGroup>
-                  {Object.keys(frontStages).map(key => (
-                    <RadioButton
-                      inline
-                      key={key}
-                      name={`${this.props.frontId} - frontStages`}
-                      checked={frontStages[key] === this.state.collectionSet}
-                      onChange={() => this.handleCollectionSetSelect(key)}
-                      label={toTitleCase(frontStages[key])}
-                    />
-                  ))}
-                </RadioGroup>
-              </StageSelectButtons>
+              {!isEditingEditions && (
+                <a
+                  href={`https://preview.gutools.co.uk/responsive-viewer/https://preview.gutools.co.uk/${this.props.frontId}`}
+                  target="preview"
+                >
+                  <FrontHeaderButton size="l">
+                    <PreviewEyeIcon size="xl" />
+                    <PreviewButtonText>Preview</PreviewButtonText>
+                  </FrontHeaderButton>
+                </a>
+              )}
+              {!isEditingEditions && (
+                <StageSelectButtons>
+                  <RadioGroup>
+                    {Object.keys(frontStages).map(key => (
+                      <RadioButton
+                        inline
+                        key={key}
+                        name={`${this.props.frontId} - frontStages`}
+                        checked={frontStages[key] === this.state.collectionSet}
+                        onChange={() => this.handleCollectionSetSelect(key)}
+                        label={toTitleCase(frontStages[key])}
+                      />
+                    ))}
+                  </RadioGroup>
+                </StageSelectButtons>
+              )}
```

Comment:
Can we have a `<EditionsOnly></EditionsOnly>` component (and inverse) that only renders when in the right state? That would reduce the boiler plate in most places I believe by moving the binding to redux inside the component?

Reply:
In #840 there's a comment that we should probably have a typed enum for the various modes the tool can be in, `fronts`, `editions`, etc.

Could possibly have a component like `<Visibility mode="fronts"> ... </Visibility>`? 

I think this should be done as a single, separate, PR, sooner rather than later though!

Reply:
Though this only solves our issues for rendering - if there's logic based on the mode, such as fetching the issue information on mount, then that'll have to be handled separately.